react-app/src/components/ListItem/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+
+import ActionIcon from 'components/ActionIcon';
+import Button from 'components/Button';
+
+import styles from './index.module.scss';
+const _classNames = classNames.bind(styles);
+
+export default class ListItem extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The date of the entry
+     */
+    date: PropTypes.string.isRequired,
+
+    /**
+     * The text of the entry
+     */
+    text: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when edit button is tapped
+     */
+    onEdit: PropTypes.func.isRequired,
+
+    /**
+     * The function to call when delete button is tapped
+     */
+    onDelete: PropTypes.func.isRequired,
+  };
+
+  state = {
+    expanded: false,
+  };
+
+  _handleEdit = () => this.props.onEdit(this.props.date);
+  _handleDelete = () => this.props.onDelete(this.props.date);
+
+  _toggleExpanded = () => this.setState({ expanded: !this.state.expanded });
+
+  _renderActionIcon() {
+    const icon = this.state.expanded ? ActionIcon.COLLAPSE : ActionIcon.EXPAND;
+
+    return <ActionIcon
+      icon={icon}
+      onClick={this._toggleExpanded}
+    />;
+  }
+
+  _renderEditButton() {
+    return (
+      <Button onClick={this._handleEdit} small>
+        Edit
+      </Button>
+    );
+  }
+
+  _renderDeleteButton() {
```

Comment:
Nothing wrong here, but unless you have to add more logic to calculate the component here, it's OK if you just declare this component with a variable inside the `render` method. Something like:

```javascript
render() {
  const deleteButton = (
    <Button onClick={this._handleDelete} small>
      Delete
    </Button>
  );

  return (
    ...
    <div>
      {deleteButton}
    </div>
    ...
  );
}
```

Reply:
Great! I will take this into account, thanks!